### PR TITLE
python3Packages.sopel: 7.0.5 -> 7.0.6 

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

Noticed it was broken in https://github.com/NixOS/nixpkgs/pull/95673#issuecomment-674837203.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
I'm currently having issues with the test suite and am looking for help. Reported the issue upstream at https://github.com/sopel-irc/sopel/issues/1926.

```
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-5.4.3, py-1.8.2, pluggy-0.13.1 -- /nix/store/fjgnz0xfl04hsblsi4ym5y5akfh6mlmy-python3-3.8.5/bin/python3.8
cachedir: .pytest_cache
rootdir: /build/sopel-7.0.6
plugins: sopel-7.0.6
collected 215 items / 1 error / 214 selected

==================================== ERRORS ====================================
_____________________ ERROR collecting sopel/test_tools.py _____________________
import file mismatch:
imported module 'sopel.test_tools' has this __file__ attribute:
  /nix/store/apl4l88a2vy1d7m6d46ncimcqgcz9xqa-python3.8-sopel-7.0.6/lib/python3.8/site-packages/sopel/test_tools.py
which is not the same as the test file we want to collect:
  /build/sopel-7.0.6/sopel/test_tools.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
=========================== short test summary info ============================
ERROR sopel/test_tools.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 1.07s ===============================
```

**Comment by drewrisinger** [comment]
@mweinelt Here's what I think is happening:

1. Nix builds sopel package
2. Nix installs sopel package to $out
3. checkPhase runs ``pytest`` in $sourceRoot
4. Pytest finds "tests" in ./sopel && ./test. "Test" in ./sopel is a mistake due to naming convention, ``test_tools.py`` registers as a file that ``pytest`` wants to run (matches ``test_*.py`` pattern). This part could be fixed by running ``pytest ./test`` in ``checkPhase``.
5. __file__ conflict is due to finding ``sopel/test_tools.py`` in two file locations: $out/.../sopel, and $sourceRoot (actually $TMP/$sourceRoot)/sopel/...

An aside, I generally recommend using ``pytestCheckHook``:

```nix
{
...
, pytestCheckHook
, ...
}:

buildPythonPackage {
  checkInputs = [ pytestCheckHook];
  pytestFlagsArray = [ "./test" ]; # prevent finding "test" in ./sopel/test_tools.py
}
```

If that doesn't work, my other suggestion is make a temp dir and move the tests to that:

```nix
(using pytestCheckHook as above)
preCheck = ''
  export TESTDIR=$(mktemp -d)
  cp -r ./test $TESTDIR
  pushd $TESTDIR
'';
postCheck = "popd";

**Comment by mweinelt** [comment]
Thanks, that explanation helped!

**Comment by mweinelt** [comment]
No space left on device :shrug:

**Comment by drewrisinger** [comment]
> No space left on device 🤷

Have you run ``nix-collect-garbage`` recently? I keep freeing up 10s of GBs using that from running ``nix-review`` all the time.

A few packages don't build with this as-is due to mocket build failures:

```
$ nix-review pr 95689:

https://github.com/NixOS/nixpkgs/pull/95689
4 packages failed to build:
python27Packages.mocket python37Packages.geoip2 python37Packages.mocket python37Packages.sopel

6 packages built:
python37Packages.maxminddb python38Packages.geoip2 python38Packages.maxminddb python38Packages.mocket python38Packages.sopel zeronet
```
Mocket might be able to be disabled on python2:
```nix
{ ...
, isPy3k
, ...
}:

buildPythonPackage {
  disabled = ! isPy3k;
}
```

Last log lines from failed mocket builds:
```
builder for '/nix/store/k34nyxsvlihrx17sd7scz84sz3df8r61-python2.7-mocket-3.8.7.drv' failed with exit code 1; last 10 log lines:
      replace_conflicting=replace_conflicting
    File "/nix/store/gz139sqlwiyycgbw1hws53q1c2v2vfa4-python2.7-setuptools-44.0.0/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1065, in best_match
      return self.obtain(req, installer)
    File "/nix/store/gz139sqlwiyycgbw1hws53q1c2v2vfa4-python2.7-setuptools-44.0.0/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1077, in obtain
      return installer(requirement)
    File "/nix/store/gz139sqlwiyycgbw1hws53q1c2v2vfa4-python2.7-setuptools-44.0.0/lib/python2.7/site-packages/setuptools/dist.py", line 777, in fetch_build_egg
      return fetch_build_egg(self, req)
    File "/nix/store/gz139sqlwiyycgbw1hws53q1c2v2vfa4-python2.7-setuptools-44.0.0/lib/python2.7/site-packages/setuptools/installer.py", line 130, in fetch_build_egg
      raise DistutilsError(str(e))
  distutils.errors.DistutilsError: Command '['/nix/store/6xyf4rb6ch0xgf6c7x2p0g6l6k76yqn2-python-2.7.18/bin/python2.7', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/build/tmpzo20Gd', '--quiet', 'pipenv']' returned non-zero exit status 1
builder for '/nix/store/pqh0as41fqx8qqa2gih7rnz1wl4lawdf-python3.7-mocket-3.8.7.drv' failed with exit code 1; last 10 log lines:
      replace_conflicting=replace_conflicting
    File "/nix/store/z00af9n2g82fiw3yqhdzhh6bm90x5b3n-python3.7-setuptools-47.3.1/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1065, in best_match
      return self.obtain(req, installer)
    File "/nix/store/z00af9n2g82fiw3yqhdzhh6bm90x5b3n-python3.7-setuptools-47.3.1/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1077, in obtain
      return installer(requirement)
    File "/nix/store/z00af9n2g82fiw3yqhdzhh6bm90x5b3n-python3.7-setuptools-47.3.1/lib/python3.7/site-packages/setuptools/dist.py", line 754, in fetch_build_egg
      return fetch_build_egg(self, req)
    File "/nix/store/z00af9n2g82fiw3yqhdzhh6bm90x5b3n-python3.7-setuptools-47.3.1/lib/python3.7/site-packages/setuptools/installer.py", line 130, in fetch_build_egg
      raise DistutilsError(str(e))
  distutils.errors.DistutilsError: Command '['/nix/store/9hdzs3g6rb76grwzlsbxyxwxg2n0fl4a-python3-3.7.8/bin/python3.7', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/build/tmpxdi3ysc3', '--quiet', 'pipenv']' returned non-zero exit status 1.
```

**Comment by mweinelt** [comment]
> Have you run nix-collect-garbage recently? I keep freeing up 10s of GBs using that from running nix-review all the time.

I'm talking about ofBorg :)

**Comment by mweinelt** [comment]
Sorted.

Result of `nixpkgs-review pr 95689` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
<br>- python27Packages.mocket
<br>- python37Packages.geoip2
<br>- python37Packages.maxminddb
<br>- python37Packages.mocket
<br>- python37Packages.sopel
<br>- python38Packages.geoip2
<br>- python38Packages.maxminddb
<br>- python38Packages.mocket
<br>- python38Packages.sopel
<br>- zeronet
</details>

**Review by jonringer — APPROVED** [review]
diff LGTM

```
https://github.com/NixOS/nixpkgs/pull/95689
10 packages built:
python27Packages.mocket python37Packages.geoip2 python37Packages.maxminddb python37Packages.mocket python37Packages.sopel python38Packages.geoip2 python38Packages.maxminddb python38Packages.mocket python38Packages.sopel zeronet
```

**Comment by jonringer** [comment]
is ofborg dead?

File: pkgs/development/python-modules/geoip2/default.nix [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
+    substituteInPlace requirements.txt --replace "requests>=2.24.0" "requests>=2.23.0"
+  '';
 
-  checkInputs = [ requests-mock ];
+  propagatedBuildInputs = [ aiohttp requests maxminddb ];
+
+  checkInputs = [ mocket requests-mock ];
 
   meta = with lib; {
     description = "MaxMind GeoIP2 API";
```
Comment (drewrisinger) [inline pkgs/development/python-modules/geoip2/default.nix:29]: Missing maintainer

File: pkgs/development/python-modules/maxminddb/default.nix [hunk 2]
```diff
@@ -7,12 +7,13 @@
...
   pname = "maxminddb";
+  disabled = pythonOlder "3.6";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "f4d28823d9ca23323d113dc7af8db2087aa4f657fafc64ff8f7a8afda871425b";
+    sha256 = "15z5557rn4yvrhnpdm9l4kczr151qv9px736hd361rlr2z98wpdr";
   };
 
   buildInputs = [ libmaxminddb ];
```
Comment (drewrisinger) [inline pkgs/development/python-modules/maxminddb/default.nix:19]: missing maintainer

File: pkgs/development/python-modules/mocket/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  version = "3.8.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1zj2p20blf4p071kcs86h2c9sc964x6sahff60gg7y4d2c0iq3rs";
+  };
+
+  patchPhase = ''
+    substituteInPlace requirements.txt --replace "python-magic==0.4.18" "python-magic"
+    substituteInPlace requirements.txt --replace "urllib3==1.25.10" "urllib3"
```
Comment (drewrisinger) [inline pkgs/development/python-modules/mocket/default.nix:20]:
```suggestion
    substituteInPlace requirements.txt \
      --replace "python-magic==0.4.18" "python-magic" \
      --replace "urllib3==1.25.10" "urllib3"
```

File: pkgs/development/python-modules/mocket/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    substituteInPlace setup.py --replace "setup_requires=[\"pipenv\"]" "setup_requires=[]"
```
Comment (drewrisinger) [inline pkgs/development/python-modules/mocket/default.nix:21]:
```suggestion
    substituteInPlace setup.py --replace 'setup_requires=["pipenv"]' "setup_requires=[]"
```

File: pkgs/development/python-modules/mocket/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+  propagatedBuildInputs = [
+    decorator
+    python_magic
+    urllib3
+    six
+  ] ++ lib.optionals (isPy27) [ six ];
+
+  # Pypi has no runtests.py, github has no requirements.txt. No way to test, no way to install.
+  doCheck = false;
```
Comment (drewrisinger) [inline pkgs/development/python-modules/mocket/default.nix:32]: Maybe file an upstream issue?

Also, it looks like the ``runtest.py`` basically just runs pytest, and the tests are included in the PyPi sdist. So maybe just try running ``pytestCheckHook``?

File: pkgs/development/python-modules/sopel/default.nix [hunk 6]
```diff
@@ -29,20 +30,26 @@ buildPythonPackage rec {
...
-    pytest .
+  preCheck = ''
+    export TESTDIR=$(mktemp -d)
+    cp -R ./test $TESTDIR
+    pushd $TESTDIR
+  '';
+
+  postCheck = ''
+    popd
   '';
```
Comment (drewrisinger) [inline pkgs/development/python-modules/sopel/default.nix:53]:
```suggestion
  postCheck = "popd";
```

File: pkgs/development/python-modules/geoip2/default.nix [hunk 7]
```diff
@@ -1,24 +1,29 @@
...
     inherit pname version;
-    sha256 = "1q1ciqqd0qjkpgwsg9fws8wcqchkcq84gv2g4q3xgh2lpj3yjsaq";
+    sha256 = "06q9r5sdmncj4yaxrdf0mls05jb5n6pwhf8j8r74825cks4mvysa";
   };
 
-  propagatedBuildInputs = [ requests maxminddb ]
-    ++ lib.optionals isPy27 [ ipaddress ];
+  patchPhase = ''
+    substituteInPlace requirements.txt --replace "requests>=2.24.0" "requests>=2.23.0"
+  '';
```
Comment (drewrisinger) [inline pkgs/development/python-modules/geoip2/default.nix:22]:
```suggestion
  postPatch= ''
    substituteInPlace requirements.txt --replace "requests>=2.24.0" "requests"
  '';
```
Personal preference is not to lock versions here at all. The tests should fail if it doesn't work with that version.

File: pkgs/development/python-modules/sopel/default.nix [hunk 8]
```diff
@@ -29,20 +30,26 @@ buildPythonPackage rec {
...
     pytz
+    sqlalchemy
     xmltodict
   ];
 
   # remove once https://github.com/sopel-irc/sopel/pull/1653 lands
   postPatch = ''
     substituteInPlace requirements.txt \
-      --replace "praw<6.0.0" "praw<7.0.0"
+      --replace "praw>=4.0.0,<6.0.0" "praw>=4.0.0,<8.0.0"
```
Comment (drewrisinger) [inline pkgs/development/python-modules/sopel/default.nix:40]:
```suggestion
      --replace "praw>=4.0.0,<6.0.0" "praw"
```
Could try not locking version and just have failed tests indicate that it doesn't work.

**Review by drewrisinger — CHANGES_REQUESTED** [review]
See comments.

File: pkgs/development/python-modules/mocket/default.nix (hunk 5 above)
Comment (jonringer) [reply pkgs/development/python-modules/mocket/default.nix:32]: generally a `tox.ini` or the `test_requires` section of setup.py will mention what is needed.

However, this does a non-standard construction. Needs `xxhash` `aiohttp` and `async-timeout`
https://github.com/mindflayer/python-mocket/blob/fb24872ae2a530dad0e645e3e9c39dcea29e999d/runtests.py#L22

File: pkgs/development/python-modules/geoip2/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/geoip2/default.nix:29]: I only noticed they were broken, I'm not going to maintain those packages.

File: pkgs/development/python-modules/maxminddb/default.nix (hunk 2 above)
Comment (mweinelt) [reply pkgs/development/python-modules/maxminddb/default.nix:19]: I only noticed they were broken, I'm not going to maintain those packages.

**Review by mweinelt — COMMENTED** [review]
> is ofborg dead?

@jonringer Yeah, graham is on vacation and cole and LnL apparently can't fix this.

**Comment by jonringer** [comment]
> @jonringer Yeah, graham is on vacation and cole and LnL apparently can't fix this.

build and test command seems to work. just eval isn't working... which I guess is what it normally does xD

File: pkgs/development/python-modules/mocket/default.nix (hunk 5 above)
Comment (mweinelt) [reply pkgs/development/python-modules/mocket/default.nix:32]: Testing mocket requires packaging of pook, which requires mocket for testing.

Boom. Infinite loop.

I only ever wanted to fixup those packages, not maintain them :disappointed:

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 95689` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
<br>- python27Packages.mocket
<br>- python37Packages.geoip2
<br>- python37Packages.maxminddb
<br>- python37Packages.mocket
<br>- python37Packages.sopel
<br>- python38Packages.geoip2
<br>- python38Packages.maxminddb
<br>- python38Packages.mocket
<br>- python38Packages.sopel
<br>- zeronet
</details>

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 95689 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>10 packages built:</summary><br>- python27Packages.mocket<br>- python37Packages.geoip2<br>- python37Packages.maxminddb<br>- python37Packages.mocket<br>- python37Packages.sopel<br>- python38Packages.geoip2<br>- python38Packages.maxminddb<br>- python38Packages.mocket<br>- python38Packages.sopel<br>- zeronet</details>

**Pull request merged**